src/main/scala/nl.knaw.dans.easy.solr4files/components/AuthenticationComponent.scala

```diff
@@ -35,6 +46,65 @@ trait AuthenticationComponent extends DebugEnhancedLogging {
       }
     }
 
-    def getUser(name: String, password: String): Try[User]
+    private def getUser(userName: String, password: String): Try[User] = {
+      // inner functions reuse the arguments
+
+      logger.info(s"looking for user [$userName]")
+
+      val query = s"(&(objectClass=easyUser)(uid=$userName))"
+      val connectionProperties = new util.Hashtable[String, String]() {
+        put(Context.PROVIDER_URL, ldapProviderUrl)
+        put(Context.SECURITY_AUTHENTICATION, "simple")
+        put(Context.SECURITY_PRINCIPAL, s"uid=$userName, $ldapUsersEntry")
+        put(Context.SECURITY_CREDENTIALS, password)
+        put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory")
+      }
+
+      def search = {
+        val searchControls = new SearchControls() {
+          setSearchScope(SUBTREE_SCOPE)
+        }
+        managed(new InitialLdapContext(connectionProperties, null))
+          .map(_.search(ldapUsersEntry, query, searchControls))
+          .tried
+      }.recoverWith {
+        case t: AuthenticationException => Failure(InvalidUserPasswordException(userName, new Exception("invalid password", t)))
+        case t => Failure(t)
+      }
+
+      def getFirst(list: List[SearchResult]): Try[SearchResult] = {
+        list
+          .headOption
+          .map(Success(_))
+          .getOrElse(Failure(InvalidUserPasswordException(userName, new Exception(s"User [$userName] not found"))))
+      }
+
+      def toTuples(a: Attribute): (String, Seq[String]) = {
+        (a.getID, a.getAll.asScala.map(_.toString).toSeq)
+      }
+
+      def userIsActive(map: Map[String, Seq[String]]): Try[Unit] = {
+        val values = map.getOrElse("dansState", Seq.empty)
+        logger.info(s"state of $userName: $values")
+        if (values.contains("ACTIVE")) Success(())
+        else Failure(InvalidUserPasswordException(userName, new Exception(s"User [$userName] found but not active")))
+      }
+
+      for {
+        searchResults <- search // returns zero or one
+        searchResult <- getFirst(searchResults.asScala.toList)
+        userAttributes = searchResult.getAttributes.getAll.asScala.map(toTuples).toMap
+        _ <- userIsActive(userAttributes)
+        roles = userAttributes.getOrElse("easyRoles", Seq.empty)
+      } yield User(userName,
+        isArchivist = roles.contains("ARCHIVIST"),
+        isAdmin = roles.contains("ADMIN"),
+        groups = userAttributes.getOrElse("easyGroups", Seq.empty)
+      )
+    }.recoverWith {
+      case t: InvalidUserPasswordException => Failure(t)
+      case t => Failure(AuthorisationNotAvailableException(t))
```

Comment:
Authorisation should be renamed into Authenticaion, 
but I'm already having merge conflicts with cleanup in #16